rpc/service.go

```diff
@@ -0,0 +1,93 @@
+package rpc
+
+import (
+	"errors"
+
+	"github.com/rs/zerolog"
+	"github.com/statechannels/go-nitro/client"
+	"github.com/statechannels/go-nitro/network"
+	netproto "github.com/statechannels/go-nitro/network/protocol"
+	"github.com/statechannels/go-nitro/protocols"
+	rpcproto "github.com/statechannels/go-nitro/rpc/protocol"
+)
+
+// TODO: logging
+
+type Service struct {
+	Logger zerolog.Logger
+
+	Client         *client.Client
+	NetworkService *network.Service
+}
+
+func NewService(cli *client.Client, nts *network.Service) *Service {
+	s := &Service{
+		Client:         cli,
+		NetworkService: nts,
+	}
+
+	go func() {
+		for {
+			peer, err := s.NetworkService.PollPeer()
+			if err != nil {
+				if errors.Is(err, network.ErrServiceClosed) {
+					s.Logger.Info().Msg("network service closed")
+
+					break
+				}
+
+				// TODO: handle error
+				s.Logger.Fatal().Err(err).Msg("failed to poll peer")
+			}
+
+			go s.HandlePeer(peer)
+		}
+	}()
+
+	return s
+}
+
+func (s *Service) HandlePeer(p *network.Peer) {
```

Comment:
Peer could be named **NetworkServiceConnection** here
**HandleNetworkServiceConnection**